main/src/library/Group.flix

```diff
@@ -0,0 +1,149 @@
+/*
+ *  Copyright 2022 Jakob Schneider Villumsen
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+///
+/// A type class for types that form a group.
+///
+/// The default instances define the additive inverse in the whole numbers.
+///
+pub lawful class Group[a] with Monoid[a] {
+
+    ///
+    /// Returns a neutral element.
+    ///
+    pub def empty(): a =
+        Monoid.empty()
+
+    ///
+    /// Returns the inverse element of `x`.
+    ///
+    pub def inverse(x: a): a
+
+    ///
+    /// Returns the result of combining `x` and `y` using Monoid's combine.
+    ///
+    pub def combine(x: a, y: a): a =
+        Monoid.combine(x, y)
+
+
+    law leftIdentity: forall(x: a) with Eq[a] . Group.combine(Group.empty(), x) == x
```

Comment:
I think some of these laws are already covered by being a SemiGroup. I think we only need laws for inverse.

Reply:
Alright. There are other type classes where laws are duplicated so that's what I did. I'll remove redundant laws from `Group` for now :)